src/platform/admin/search/admin.search.ingest.resolver.mutations.ts

```diff
@@ -41,27 +48,67 @@ export class AdminSearchIngestResolverMutations {
       AuthorizationPrivilege.PLATFORM_ADMIN,
       `Ingest new data into Elasticsearch from scratch: ${agentInfo.email}`
     );
-    const deleteResult = await this.searchIngestService.removeIndices();
 
-    if (!deleteResult.acknowledged) {
-      throw new BaseException(
-        deleteResult.message ?? 'Failed to delete indices',
-        LogContext.SEARCH_INGEST,
-        AlkemioErrorStatus.UNSPECIFIED
-      );
-    }
+    this.logger.verbose?.('Starting search ingest from scratch');
+
+    const task = await this.taskService.create();
+    // todo: add error handling
+    this.searchIngestService
+      .removeIndices()
+      .then(res => {
+        if (!res.acknowledged) {
+          throw new Error(
+            res.message ? res.message : 'Failed to delete indices'
+          );
+        }
+        return this.searchIngestService.ensureIndicesExist();
+      })
+      .then(res => {
+        if (!res.acknowledged) {
+          throw new Error(
+            res.message ? res.message : 'Failed to create indices'
+          );
+        }
+        return this.searchIngestService.ingest(task);
+      })
+      .then(() => this.taskService.complete(task.id))
+      .catch(async e => {
+        await this.taskService.updateTaskErrors(task.id, e?.message);
+        return this.taskService.complete(task.id, TaskStatus.ERRORED);
+      });
+    // const deleteResult = await this.searchIngestService.removeIndices();
```

Comment:
Cleanup / raise story.

Reply:
No longer needed. The progress is handled in the task.